C builderr with option as struct field.

### Describe the bug

When setting a struct field to an option with a default value a C error occurs.

### Reproduction Steps

```v
type SomeType = SomeStruct | string

struct SomeStruct {}

struct AnotherStruct {
	field ?SomeType = 'default_string'
}

fn main() {
	println(AnotherStruct{})
}

```

### Expected Behavior

Expected code to compile and print this:
```
AnotherStruct{
    field: Option('default_string')
}
```

### Current Behavior

```
C:/Users/imado/AppData/Local/Temp/v_0/test.2663439158810552116.tmp.c:507: warning: WINVER redefined
C:/Users/imado/AppData/Local/Temp/v_0/test.2663439158810552116.tmp.c:6907: warning: implicit declaration of function 'tcc_backtrace'
C:/Users/imado/AppData/Local/Temp/v_0/test.2663439158810552116.tmp.c:12679: error: cannot convert 'struct string' to 'unsigned char'
builder error: 
==================
C error. This should never happen.
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V full version: V 0.4.1 68cbf27.3fab321

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.1 68cbf27.3fab321
OS: windows, Microsoft Windows 11 Pro v22621 64-bit
Processor: 16 cpus, 64bit, little endian, 

getwd: C:\Users\imado\Documents\cyberian_tiger
vexe: C:\Users\imado\v\v.exe
vexe mtime: 2023-09-23 02:25:12

vroot: OK, value: C:\Users\imado\v
VMODULES: OK, value: C:\Users\imado\.vmodules
VTMP: OK, value: C:\Users\imado\AppData\Local\Temp\v_0

Git version: git version 2.33.1.windows.1
Git vroot status: weekly.2023.38-24-g3fab3213
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,
operable program or batch file.

thirdparty/tcc status: thirdparty-windows-amd64 e90c2620
```

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -359,8 +359,7 @@ fn (mut g Gen) zero_struct_field(field ast.StructField) bool {
 	if field.has_default_expr {
 		if sym.kind in [.sum_type, .interface_] {
 			if field.typ.has_flag(.option) {
-				g.expr_opt_with_cast(field.default_expr, field.default_expr_typ.set_flag(.option),
-					field.typ)
+				g.expr_with_opt(field.default_expr, field.default_expr_typ, field.typ)
 			} else {
 				g.expr_with_cast(field.default_expr, field.default_expr_typ, field.typ)
 			}

--- a/vlib/v/tests/struct_field_option_type_with_default_value_init_test.v
+++ b/vlib/v/tests/struct_field_option_type_with_default_value_init_test.v
@@ -1,0 +1,13 @@
+type SomeType = SomeStruct | string
+
+struct SomeStruct {}
+
+struct AnotherStruct {
+	field ?SomeType = 'default_string'
+}
+
+fn test_struct_field_option_type_with_default_value() {
+	s := AnotherStruct{}
+	println(s.field)
+	assert '${s.field}' == "Option(SomeType('default_string'))"
+}

